iDiff/cmd/root.go

```diff
@@ -1,22 +1,101 @@
 package cmd
 
 import (
+	"bytes"
+	"errors"
 	goflag "flag"
+	"fmt"
+	"os"
+	"regexp"
 
+	"github.com/GoogleCloudPlatform/runtimes-common/iDiff/differs"
 	"github.com/golang/glog"
 	"github.com/spf13/cobra"
 	"github.com/spf13/pflag"
 )
 
+var json bool
+
 var RootCmd = &cobra.Command{
-	Use:   "iDiff",
-	Short: "iDiff is an image differ tool.",
-	Long:  `iDiff is an image differ tool.`,
-	Run: func(command *cobra.Command, args []string) {
-		glog.Info("Root command started")
+	Use:   "iDiff [differ] [container1] [container2]",
+	Short: "Compare two images.",
+	Long:  `Compares two images using the specifed differ (see iDiff documentation for available differs).`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if validArgs, err := validateArgs(args[1:]); !validArgs {
+			glog.Error(err.Error())
+			os.Exit(1)
+		}
+		if diff, err := differs.Diff(args[2], args[3], args[1], json); err == nil {
+			fmt.Println(diff)
+		} else {
+			glog.Error(err.Error())
+		}
 	},
 }
 
+func validateArgs(args []string) (bool, error) {
+	if validArgNum, err := checkArgNum(args); !validArgNum {
```

Comment:
Check the error here as well.  It doesn't matter currently but the implementation for validArgNum might change.